ironfish/src/rpc/clients/tcpClient.ts

```diff
@@ -36,20 +36,21 @@ export class IronfishTcpClient extends IronfishRpcClient {
   }
 
   async connect(): Promise<void> {
-    const connected = await this.connectClient()
-      .then(() => true)
-      .catch(() => false)
+    let connected = false
+    await this.connectClient()
+      .then(() => (connected = true))
+      .catch((error) => {
+        if (!this.retryConnect) {
+          this.logger.error(`Failed to connect to ${String(this.host)}:${String(this.port)}`)
+          throw error
+        }
+      })
 
     if (!connected) {
-      if (this.retryConnect) {
-        this.logger.warn(
-          `Failed to connect to ${String(this.host)}:${String(this.port)}, retrying`,
-        )
-        this.connectTimeout = setTimeout(() => void this.connect(), CONNECT_RETRY_MS)
-        return
-      }
-
-      this.logger.warn(`Failed to connect to ${String(this.host)}:${String(this.port)}`)
```

Comment:
Does the code need to change? Should we just throw at line 53? I think it's hard for me to understand why the code structure got more complex without an explanation.

Reply:
We could throw at line 53, but I think we'd want to throw the error that we catch from `connectClient`. I thought that this was simpler than what I came up with for storing that error and propagating it, but I could give it another pass

Reply:
This looks good, thanks for the explanation.